### ISSUE TITLE
Irritating error message about IPv6 authentication

### ISSUE THREAD

**Issue by henk84** [rsmarples/dhcpcd#143]
Hi,

I’m getting the following messages in my syslog:
```
Oct 26 14:18:47 frustcomp dhcpcd[12572]: eno0: authentication failed from fe80::20d:b9ff:fe21:b964: No such file or directory
```
The appear in clusters of 8 over something like 3-5 minutes, with a gap of 10-60 minutes between the clusters.

I have no idea what that error is telling me. So

1. I’m asking for help with understanding and fixing the issue.
2. I suggest rephrasing the error message to give a bit more of an indication what this is about, maybe even what the cause might be and how to fix it or how to tell dhcpcd to ignore that.

My system is Debian stable/bullseye, i.e. the package name is dhcpcd5 in version 7.1.0-2+b1.
The system is running sysvinit, not systemd (that may be relevant WRT temporary file locations or so, so I thought I should mention it upfront.)

Concrete questions that might help me understand and fix this:
What file is it looking for?
What authentication is this? Does it come from the DHCP server?

The only thing I tried so far is placing `noauthrequired` in `/etc/dhcpcd.conf` but that didn’t change the behaviour.

Any advice how to solve this issue is appreciated, thank you!

**Comment by rsmarples** [comment]
DHCP authentication is tricky. Anyway, it looks like you have *not* setup any DHCP authentication at your end? Looks like you received a DHCP message with authentication and the error is trying to say you don't have any *that matches*.

The actual error is here: https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/auth.c#L281
That means someone is sending you a DHCP message to renew your config. That *requires* authentication.

Can you think of a better errno to return than ENOENT?

Anyway, there might be a bug with the auto authentication dhcpcd supports for that - dhcpcd-7 is quite old.
Can you try dhcpcd-9 please?

**Comment by henk84** [comment]
> DHCP authentication is tricky. Anyway, it looks like you have _not_ setup any DHCP authentication at your end?

I don’t think so. I wasn’t even aware of the DHCP having any authentication …
The DHCP server (with the IPv6 link-local address mentioned in the error message) is running on OpenWrt 19.07.3 r11063-85e04e9f46.
AFAICT this uses dnsmasq and odhcpd-ipv6, at least these packages are installed:
dnsmasq 2.80-16.1
odhcpd-ipv6only 2020-05-03-49e4949c-3

I’m not aware of any options I might have changed enabling dhcp authentication.
Neither on the client machine that’s throwing these errors.

> Looks like you received a DHCP message with authentication and the error is trying to say you don't have any _that matches_.
> 
> The actual error is here: https://github.com/NetworkConfiguration/dhcpcd/blob/master/src/auth.c#L281 That means someone is sending you a DHCP message to renew your config. That _requires_ authentication.

That someone is probably the DHCP server on my OpenWRT, right?

> Can you think of a better errno to return than ENOENT?

No, I don’t really know anything about errnos, not even really what that term means (I’m just a sysadmin and never had to deal with those).
I just know that an error only saying "no such file" without saying what file (and AFAIU not even really meaning a file at all) and providing more info on the issue and how to solve it, is bad UX design because it does not mean anything to the person who reads it without reading the code that produces it.

> Anyway, there might be a bug with the auto authentication dhcpcd supports for that - dhcpcd-7 is quite old. Can you try dhcpcd-9 please?

Hm, not easily. There is no dhcpcd-9 available for Debian bullseye/stable, only for Debian bookworm/testing.
And I don’t currently have a machine running that or a possibility to set that up without investing more time than I have at disposal currently.
Do you have reason to believe it is fixed in dhcpcd-9 or is this a shot in the dark?

**Comment by henk84** [comment]
> > Anyway, there might be a bug with the auto authentication dhcpcd supports for that - dhcpcd-7 is quite old. Can you try dhcpcd-9 please?
> Hm, not easily. There is no dhcpcd-9 available for Debian bullseye/stable, only for Debian bookworm/testing.

FWIW: I have updated to bookworm after it became stable and now have:
```
ii  dhcpcd         9.4.1-22     all          DHCPv4 and DHCPv6 dual-stack client (init.d script and systemd unit)
```

The error still occurs:
```
2023-09-07T14:04:36.570274+02:00 frustcomp dhcpcd[8162]: eno0: authentication failed from fe80::20d:b9ff:fe21:b964: No such file or directory
```

**Comment by perkelix** [comment]
Does the above still apply to 10.5.2?
